sql_generators/glean_usage/glean_app_ping_views.py

```diff
@@ -256,3 +261,18 @@ def _generate_select_expression(
                         select_expr.append(f"SAFE_CAST(NULL AS {dtype}) AS {node_name}")
 
         return ", ".join(select_expr)
+
+    def _type_info(self, nodes):
+        """Determine the type information"""
+        type_expr = []
+        for node in nodes:
+            dtype = node["type"]
+            if dtype == "RECORD":
+                if node.get("mode", None) == "REPEATED":
+                    type_expr.append(f"ARRAY<STRUCT<{self._type_info(node['fields'])}>>")
+                else:
+                    type_expr.append(f"STRUCT<{self._type_info(node['fields'])}>")
+            else:
+                type_expr.append(dtype)
```

Comment:
Need to handle possible arrays in this logic branch as well.
```suggestion
                if node.get("mode", None) == "REPEATED":
                    type_expr.append(f"ARRAY<{dtype}>")
                else:
                    type_expr.append(dtype)
```